### DISCUSSION TITLE
nslookup timedout when use browsing security and parent control service in General Settings.

### DISCUSSION THREAD

**Discussion by wxf26054** [AdguardTeam/AdGuardHome#2742] [Q&A]
![image](https://user-images.githubusercontent.com/22016320/107872047-5143e380-6ee2-11eb-88d5-e989030023b2.png)
These are errors in the log output.
![image](https://user-images.githubusercontent.com/22016320/107872088-8b14ea00-6ee2-11eb-9b36-7bc96f9bcdd0.png)

**Comment by ameshkov** [comment]
Well, it just means that `dns-family.adguard.com` is not reachable from your place.

Could it be your ISP blocking it?

**Comment by wxf26054** [comment]
No, It shows ping value is high, but not blocked by the ISP.
![image](https://user-images.githubusercontent.com/22016320/107916762-ce825d80-6fa1-11eb-8e58-695f8b915506.png)

**Comment by ameshkov** [comment]
Well, pinging is not the same as making a query, also packet loss is quite high.

Try using [dnslookup](https://github.com/ameshkov/dnslookup) and do some simple DNS queries:
`dnslookup example.org https://dns-family.adguard.com/dns-query`

**Comment by wxf26054** [comment]
It has normal Query answer.
![image](https://user-images.githubusercontent.com/22016320/108647052-a4292680-74f2-11eb-8236-c916f2167832.png)

**Reply by ameshkov** [reply]
How long does it take to respond?

**Reply by wxf26054** [reply]
It seems normal, maybe this problem is caused by China's GFW service.

**Reply by ameshkov** [reply]
Yeah, this may be the reason indeed, GFW may limit HTTPS connections.

**Comment by ppfeufer** [comment]
Got the same issue, enabling browsing security results in AdGuardHome being unresponsive after a while and blocking network traffic

```
;; QUESTION SECTION:
;discord.com.   IN       A

2021/12/08 06:03:07.469947 1009#29196 [debug] github.com/AdguardTeam/AdGuardHome/internal/dnsfilter.(*DNSFilter).checkSafeBrowsing(): SafeBrowsing lookup for discord.com; Elapsed time: 3001ms
2021/12/08 06:03:07.470253 1009#29196 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).udpHandlePacket(): error handling DNS (udp) request: talking to dnsUpstream failed, cause: dnsfilter failed to check host "discord.com": safe browsing: couldn't do a GET request to 'https://dns-family.adguard.com:443/dns-query', cause: Get "https://dns-family.adguard.com:443/dns-query?dns=IoMBAAABAAAAAAAABGM4ZTgCc2IDZG5zB2FkZ3VhcmQDY29tAAAQAAE": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2021/12/08 06:03:07.470422 1009#29319 [debug] using settings for client [PC] thor with ip 192.168.178.20 and id ""
2021/12/08 06:03:07.470712 1009#29319 [debug] etchostscontainer: answer: discord.com -> []
2021/12/08 06:03:07.472546 1009#29319 [debug] github.com/AdguardTeam/AdGuardHome/internal/dnsfilter.check(): SafeBrowsing: checking discord.com: c8e8.sb.dns.adguard.com.
2021/12/08 06:03:07.473309 1009#29319 [debug] https://dns-family.adguard.com:443/dns-query: sending request TXT c8e8.sb.dns.adguard.com.
```

dnslookup:
```
$ ./dnslookup example.org https://dns-family.adguard.com/dns-query
dnslookup v. v1.5.1
dnslookup result:
;; opcode: QUERY, status: NOERROR, id: 43372
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;example.org.   IN       A

;; ANSWER SECTION:
example.org.    12218   IN      A       93.184.216.34
```
